detox/android/detox/src/main/java/com/wix/detox/DetoxActionHandlers.kt

```diff
@@ -110,3 +110,61 @@ class QueryStatusActionHandler(
             })
         }
 }
+
+class RecordingStateActionHandler(
+        private val instrumentsManager: DetoxInstrumentsManager,
+        private val wsClient: WebSocketClient
+) : DetoxActionHandler {
+
+    override fun handle(params: String, messageId: Long) {
+        val recordingPath = JSONObject(params).optString("recordingPath", null)
+        if (recordingPath != null) {
+            instrumentsManager.startRecordingAtLocalPath(recordingPath)
+        } else {
+            instrumentsManager.stopRecording()
+        }
```

Comment:
While I'm ok with the code, overall, I don't entirely understand the contract here. The name `recordingPath` lacks the meaning that the recording starts; The omission of it lacks the meaning that the recording stops.
I think you guys need to rethink this protocol -- namely, make it more explicit (e.g. record="start"/"stop" alongside recordPath as a parameter when record="start")

Reply:
@d4vidi Protocol is the same as for iOS as well.

Reply:
@d4vidi, Valentyn is right. That is what was implemented for iOS.

Reply:
Would you consider changing this?

Reply:
Not for this iteration. As then we'll need to change the iOS part as well.